File: articles/implementation-pathways-and-strategic-sequencing/cpp/sequencing_score.cpp
// Efficient sequencing readiness scoring example.
// Compile: g++ sequencing_score.cpp -std=c++17 -O2 -o sequencing_score
// Run: ./sequencing_score

#include <iostream>
#include <string>
#include <vector>

struct Pathway {
    std::string name;
    double capability, evidence, governance, legitimacy, dependency, reversibility, capacity, timing, ethics;
};

double score(const Pathway& p) {
    return 0.16*p.capability + 0.15*p.evidence + 0.15*p.governance + 0.14*p.legitimacy -
           0.12*p.dependency + 0.10*p.reversibility - 0.10*p.capacity + 0.08*p.timing + 0.12*p.ethics;
}

int main() {
    std::vector<Pathway> pathways = {
        {"Data Governance Foundation", 0.72, 0.66, 0.78, 0.62, 0.42, 0.70, 0.46, 0.52, 0.70},
        {"Full Platform Rollout", 0.52, 0.46, 0.48, 0.44, 0.82, 0.38, 0.84, 0.60, 0.42},
        {"Adaptive Rollout Sequence", 0.70, 0.68, 0.72, 0.74, 0.52, 0.76, 0.60, 0.66, 0.78}
    };
    for (const auto& p : pathways) std::cout << p.name << " | sequencing readiness " << score(p) << "\n";
    return 0;
}

File: articles/implementation-pathways-and-strategic-sequencing/rust/sequencing_score.rs
// Command-line sequencing readiness scoring scaffold.
// Compile: rustc sequencing_score.rs -o sequencing_score
// Run: ./sequencing_score

struct Pathway {
    name: &'static str,
    capability: f64,
    evidence: f64,
    governance: f64,
    legitimacy: f64,
    dependency: f64,
    reversibility: f64,
    capacity: f64,
    timing: f64,
    ethics: f64,
}

fn sequencing_readiness(p: &Pathway) -> f64 {
    0.16 * p.capability
        + 0.15 * p.evidence
        + 0.15 * p.governance
        + 0.14 * p.legitimacy
        - 0.12 * p.dependency
        + 0.10 * p.reversibility
        - 0.10 * p.capacity
        + 0.08 * p.timing
        + 0.12 * p.ethics
}

fn main() {
    let pathways = vec![
        Pathway { name: "Data Governance Foundation", capability: 0.72, evidence: 0.66, governance: 0.78, legitimacy: 0.62, dependency: 0.42, reversibility: 0.70, capacity: 0.46, timing: 0.52, ethics: 0.70 },
        Pathway { name: "Full Platform Rollout", capability: 0.52, evidence: 0.46, governance: 0.48, legitimacy: 0.44, dependency: 0.82, reversibility: 0.38, capacity: 0.84, timing: 0.60, ethics: 0.42 },
        Pathway { name: "Adaptive Rollout Sequence", capability: 0.70, evidence: 0.68, governance: 0.72, legitimacy: 0.74, dependency: 0.52, reversibility: 0.76, capacity: 0.60, timing: 0.66, ethics: 0.78 },
    ];

    for p in pathways {
        println!("{} | sequencing readiness {:.3}", p.name, sequencing_readiness(&p));
    }
}
